### PR TITLE
GHA: update release action

### PR DESCRIPTION
**Pull request by martinfleis**
Follow `spaghetti` example and generate changelog using Github.

### CODE REVIEW

File: .github/workflows/release_and_publish.yml [hunk 1]
```diff
@@ -1,65 +1,57 @@
...
        - name: Set up python
          uses: actions/setup-python@v4
          with:
            python-version: "3.x"
- 
+
        - name: Install Dependencies
          run: |
            python -m pip install --upgrade pip
-           pip install setuptools wheel twine jupyter urllib3 pandas pyyaml
```
Comment (martinfleis) [inline .github/workflows/release_and_publish.yml:34]: My assumption is that `jupyter urllib3 pandas pyyaml` were required for `gitcount`

Comment (jGaboardi) [reply .github/workflows/release_and_publish.yml:34]: Good point. Let's try without. Then can update other packages when we know for sure.

**Review by jGaboardi — APPROVED** [review]

**Comment by jGaboardi** [comment]
xref https://github.com/pysal/libpysal/issues/562

File: .github/workflows/release_and_publish.yml (hunk 1 above)
Comment (knaaptime) [reply .github/workflows/release_and_publish.yml:34]: i think thats right

**Pull request merged**